gatsby/content/blog/2020/04/2020-04-06-selfhosting-matrix.mdx

```diff
@@ -0,0 +1,162 @@
+fresh Debian install.  It's (almost) filmed in a single shot, and takes about
+20 minutes from beginning to end.
+
+Please note that this does assume you're pretty familiar with Linux system
+administration.  If you're not, then we'd recommend using a Matrix hosting
+provider such as [Modular.im](https://modular.im) (which directly supports development of the core team),
+[Ungleich.ch](https://ungleich.ch/u/products/hosted-matrix-chat/), or [StartupStack](https://startupstack.tech/).
+
+Finally, while the video shows how to install on Debian via Debian packages,
+there are many many other environments and architectures (e.g. installing
+under Docker) - this is just one relatively easy way to skin the cat.  Perhaps
+there will be other 'speed-run' videos in future :)
+
+<iframe width="1229" height="691" src="https://www.youtube.com/embed/dDddKmdLEdg" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+If you want to follow along at home without listening to the video (and I can't blame you if you do ;) the high level steps are as follows:
+
+### Debian & DNS
+
+ * Take one fresh Debian 10 install.
+ * Point the DNS for your domain to it.  You should use separate subdomains for the various services as a hygiene measure to make cross-site scripting attacks less effective.  In this example, we set up DNS for:
+   * `dangerousdemos.net` (general website, and for hosting a .well-known path to advertise the Matrix service)
+   * `matrix.dangerousdemos.net` (Synapse)
+   * `riot.dangerousdemos.net` (Riot/Web)
+   * `jitsi.dangerousdemos.net` (Jitsi video conferencing)
+   * In practice, we used a `*.dangerousdemos.net` wildcard DNS record for the three subdomains in this instance.
+
+### Nginx and LetsEncrypt
+
+ * Install nginx as a webserver: `apt-get update && apt -y install nginx`
+ * Go to `/etc/nginx/sites-enabled` and copy the vhost configuration block from the bottom of `default` to new files called `dangerousdemos.net`, `matrix.dangerousdemos.net`, and `riot.dangerousdemos.net`.
+   * Rename the `server_name` field in the new files to match the hostname of each host, and point `root` to an appropriate location per domain (e.g. `/var/www/dangerousdemos.net`)
+   * For the Synapse domain (`matrix.dangerousdemos.net` here), you should replace the contents of the `location` block with `proxy_pass http://localhost:8008;` - telling nginx to pass the traffic through to synapse, which listens by default for plaintext HTTP traffic on port 8008.  (N.B. do **not** put a trailing slash on the URL here, otherwise nginx will mangle the forwarded URLs.)
+ * Enable TLS via LetsEncrypt on nginx, by: `apt install -y python3-certbot-nginx && certbot --nginx -d dangerousdemos.net -d riot.dangerousdemos.net -d matrix.dangerousdemos.net` (or whatever your domains are).
+ * You should be able to go to https://dangerousdemos.net at this point and see a page with valid HTTPS.
+
+### Synapse
```

Comment:
Probably just nitpicking, but it might be nice to briefly mention that `INSTALL.md` features some alternative methods for non-Debian setups, wdyt?